execution/gethexec/timeboost_sequencer.go

```diff
@@ -0,0 +1,555 @@
+	f.Int(prefix+".nonce-cache-size", DefaultTimeboostSequencerConfig.NonceCacheSize, "size of the tx sender nonce cache")
+	f.Uint64(prefix+".max-revert-gas-reject", DefaultTimeboostSequencerConfig.MaxRevertGasReject, "maximum gas executed in a revert for the sequencer to reject the transaction instead of posting it (anti-DOS)")
+	f.Duration(prefix+".parent-chain-finalization-time", DefaultTimeboostSequencerConfig.ParentChainFinalizationTime, "parent chain finalization time")
+	f.Duration(prefix+".max-acceptable-timestamp-delta", DefaultTimeboostSequencerConfig.MaxAcceptableTimestampDelta, "maximum acceptable time difference between the local time and the latest L1 block's timestamp")
+	f.Bool(prefix+".enable-profiling", DefaultTimeboostSequencerConfig.EnableProfiling, "enable CPU profiling and tracing")
+	f.Duration(prefix+".metric-time-for-block-creation", DefaultTimeboostSequencerConfig.MetricTimeForBlockCreation, "time to measure the time it takes to create a block")
+	TimeboostListenerConfigAddOptions(prefix+".timeboost-listener-config", f)
+}
+
+func NewTimeboostSequencer(execEngine *ExecutionEngine, l1Reader *headerreader.HeaderReader, configFetcher TimeboostSequencerConfigFetcher) (*TimeboostSequencer, error) {
+	timeboostListener := &TimeboostListener{
+		config: configFetcher().TimeboostListenerConfig,
+		conn:   nil,
+	}
+	return &TimeboostSequencer{
+		config:               configFetcher,
+		execEngine:           execEngine,
+		l1Reader:             l1Reader,
+		nonceCache:           newNonceCache(configFetcher().NonceCacheSize),
+		timeboostTxnListener: timeboostListener,
+	}, nil
+}
+
+func (s *TimeboostSequencer) createBlock(ctx context.Context) (returnValue bool) {
+
+	// First we need to create the current list of transactions that we will process
+	queueItems := make([]timeboostTransactionQueueItem, 0)
+	var totalBlockSize int
+	madeBlock := false
+
+	defer func() {
+		panicErr := recover()
+		if panicErr != nil {
+			log.Error("sequencer block creation panicked", "panic", panicErr, "backtrace", string(debug.Stack()))
+			// TODO: we should return errors to the user here
+			// For now we are logging
+			for _, queueItem := range queueItems {
+				log.Error("Error processing transaction", "err", sequencerInternalError, "queueItem", queueItem)
+			}
+		}
+
+		returnValue = true
+	}()
+
+	lastBlock := s.execEngine.bc.CurrentBlock()
+	config := s.config()
+
+	for {
+		var queueItem timeboostTransactionQueueItem
+		//  Transaction retry queue should only
+		//  have transactions from a given round id
+		if s.txRetryQueue.Len() > 0 {
+			queueItem = s.txRetryQueue.deQueue()
+		} else if s.txQueue.Len() == 0 {
+			// This means we have no transactions in the txRetryQueue and
+			// we also dont have any sailfish rounds to process
+			break
+		} else {
+			// Only add transactions from the same round id or if the queue is empty
+			if len(queueItems) == 0 {
```

Comment:
Wait isn't this `queueItems` always empty at this point in the function?

Reply:
why would it always be empty? every iteration we are hoping to add a new queueItem to the list

Reply:
Sorry I didn't see line `219`